### PR TITLE
feat(Banner): add banner component

### LINKED ISSUE #4244: Introduce banner component and demo
Placeholder for when https://github.com/patternfly/patternfly/issues/2727 is complete

### PR DESCRIPTION
**Pull request by kmcfaul**
**What**: Closes #4244 

Also includes core version bump to 4.12.0 for banner styling

### CODE REVIEW

File: packages/react-core/src/components/Banner/__tests__/Banner.test.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Banner } from '../Banner';
+import React from 'react';
+import { mount } from 'enzyme';
+
+['default', 'info', 'success', 'warning', 'danger'].forEach(
+  (variant: 'default' | 'info' | 'success' | 'warning' | 'danger') => {
```
Comment (nicolethoen) [inline packages/react-core/src/components/Banner/__tests__/Banner.test.tsx:6]: Probably worth testing the isSticky prop?

**Comment by kmcfaul** [comment]
Sure I'll add a demo & test for `isSticky`

File: packages/react-integration/demo-app-ts/src/components/demos/BannerDemo/BannerDemo.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { Banner } from '@patternfly/react-core';
+
+export class BannerDemo extends React.Component {
+  static displayName = 'BannerDemo';
+  componentDidMount() {
+    window.scrollTo(0, 0);
+  }
+
+  render() {
```
Comment (tlabaj) [inline packages/react-integration/demo-app-ts/src/components/demos/BannerDemo/BannerDemo.tsx:10]: Can you test the isSticky prop on one of these Banners?

File: packages/react-core/src/components/Banner/Banner.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  children?: React.ReactNode;
+  /** Additional classes added to the banner */
+  className?: string;
+  /** Variant styles for the banner */
+  variant?: 'default' | 'info' | 'danger' | 'success' | 'warning';
+  /** If set to true, the banner sticks to the top of its container */
+  isSticky?: boolean;
+}
+
+const variantStyle = {
```
Comment (tlabaj) [inline packages/react-core/src/components/Banner/Banner.tsx:16]: rather than having this const can we use the css 'getModifier` function?

Comment (tlabaj) [reply packages/react-core/src/components/Banner/Banner.tsx]: Sorry.  Did not realize that had been removed.  Let's follow what is done in Alerts.

File: packages/react-core/src/components/Banner/Banner.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+export const Banner: React.FunctionComponent<BannerProps> = ({
+  children,
+  className,
+  variant = 'default',
+  isSticky = false
+}: BannerProps) => (
+  <div
+    className={css(
+      styles.banner,
+      styles.modifiers[variant as 'success' | 'danger' | 'warning' | 'info'],
```
Comment (tlabaj) [inline packages/react-core/src/components/Banner/Banner.tsx:25]: do we need to check for `variant !== default` here?

Comment (kmcfaul) [reply packages/react-core/src/components/Banner/Banner.tsx:25]: If `variant === default` no class gets applied, so I don't think we need to check for it.

File: packages/react-core/src/demos/PageLayout/examples/BannerDemo.md [hunk 5]
```diff
@@ -0,0 +1,528 @@
...
+          mainContainerId={pageId}
+        >
+          <Banner isSticky>
+            <Flex
+              justifyContent={{ default: 'justifyContentCenter', lg: 'justifyContentSpaceBetween' }}
+              flexWrap={{ default: 'nowrap' }}
+            >
+              <div className="pf-u-display-none pf-u-display-block-on-lg">Localhost</div>
+              <div className="pf-u-display-none pf-u-display-block-on-lg">
+                This message is sticky to the top or bottom of the page.
```
Comment (nicolethoen) [inline packages/react-core/src/demos/PageLayout/examples/BannerDemo.md:242]: I think it's just a little confusing that the two demos, which appear to have banners sticking to three different parts of the page all have the same message that says, "This message is sticky to the top or bottom of the page." Maybe the messages should each be a little more specific and different?

Comment (kmcfaul) [reply packages/react-core/src/demos/PageLayout/examples/BannerDemo.md:242]: That's how they are in the core demo. I can update them to reference their specific location.

Comment (mcoker) [reply packages/react-core/src/demos/PageLayout/examples/BannerDemo.md]: Agree, we should update that in the core demo, too. If you want to make that change in this demo, core can update to match.

**Pull request merged**